### PR TITLE
Add ability to pull dead body out of vehicle

### PR DESCRIPTION
**Pull request by Dystopian**
**When merged this pull request will:**
- title;
- disable interaction with crew of dead vehicle;
- disable `Unload patient` medical action when he's dead.

Action is added to `Passengers` submenu.
Getting in right seat is used to pull out dead body - looks similar to RL.
Tested also on RHS vehicles with openable doors.

As usually English mistake remarks are welcomed.
Thanks to #5626, old dreams come true.

### CODE REVIEW

File: addons/interaction/functions/fnc_canPullOutBody.sqf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+params ["_target", "_unit"];
+
+private _vehicle = objectParent _target;
+
+if (
+    alive _target
+    || {_target == _vehicle}
+    || {1 < locked _vehicle}
+    || {
+        0 < {alive _x} count crew _vehicle // alive is in vehicle
```
Comment (PabstMirror) [inline addons/interaction/functions/fnc_canPullOutBody.sqf]: `<=` ?

**Comment by PabstMirror** [comment]
Could also use a dummy agent to remove the body, which would skip the need for the player to get in.

```
private _dummy = createAgent ["C_UAV_AI_F", (getPos player), [], 0, "NONE"];
// first get in to target seat
if (!(_turretPath isEqualTo [])) then {
    _dummy moveInTurret [_vehicle, _turretPath];
} else {
    if (_cargoIndex > -1) then {
        _dummy moveInCargo [_vehicle, _cargoIndex];
    } else {
        _dummy moveInDriver _vehicle;
    };
};

[
{
    params ["_dummy", "_vehicle"];
    (isNull _vehicle) || {isNull _dummy} || {_dummy in _vehicle}
},
{
    params ["_dummy", "_vehicle"];
    TRACE_1("deleting normally",_vehicle);
    deleteVehicle _dummy;
},
[_dummy, _vehicle],
3,
{
    params ["_dummy", "_vehicle"];
    TRACE_1("timeout",_vehicle);
    deleteVehicle _dummy;
}] call CBA_fnc_waitUntilAndExecute;
```

**Comment by commy2** [comment]
How do you know it would push out __this__ body and not another person?

**Comment by Dystopian** [comment]
@PabstMirror yes, but it will look more weird without animation, and maybe progress bar will be required. The way I implemented is more similar to vanilla one.

@commy2 I take a seat of this body so it's pulled out.

File: addons/interaction/functions/fnc_pullOutBody.sqf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+private _preserveEngineOn = false;
+
+// first get in to target seat
+if (!(_turretPath isEqualTo [])) then {
+    _unit action ["GetInTurret", _vehicle, _turretPath];
+} else {
+    if (_cargoIndex > -1) then {
+        _unit action ["GetInCargo", _vehicle, _cargoIndex];
+    } else {
+        _unit action ["GetIn" + _role, _vehicle];
```
Comment (Dystopian) [inline addons/interaction/functions/fnc_pullOutBody.sqf]: replace with `GetInDriver`

File: addons/interaction/functions/fnc_pullOutBody.sqf [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+TRACE_3("found crew",_role,_cargoIndex,_turretPath);
+
+private _preserveEngineOn = false;
+
+// first get in to target seat
+if (!(_turretPath isEqualTo [])) then {
+    _unit action ["GetInTurret", _vehicle, _turretPath];
+} else {
+    if (_cargoIndex > -1) then {
+        _unit action ["GetInCargo", _vehicle, _cargoIndex];
```
Comment (Dystopian) [inline addons/interaction/functions/fnc_pullOutBody.sqf]: error here and in quickmount, `_cargoIndex` must be replaced with cargo order number, [see Description](https://community.bistudio.com/wiki/fullCrew)

**Comment by Dystopian** [comment]
Sorry, was busy with another PR. Now fixed.

**Pull request merged**